mountaineer/dependencies/core/__init__.py

```diff
@@ -0,0 +1,5 @@
+"""
+Core dependencies for Mountaineer projects.
+
+"""
+from .core import get_config_with_type as get_config_with_type
```

Comment:
One of the bigger changes that pyright necessitated. It wasn't able to sniff our previous DependencyBase definitions for the right function signature of dependencies, since the functions weren't decorated with any kind of `@classmethod` or `@staticmethod`. As addressed in https://github.com/piercefreeman/mountaineer/pull/39, we can't actually use staticmethods because they interfere with our recursive dependency resolution logic.

We still want to be able to provide clients with a class-like interface that allows method introspection to better advertise what dependencies are available. Our solution here is to add and dynamically rename a limited scope Python module, like here in this `__init__.py` file. We call this `CoreDependencies` in the next level of dependencies, which allows client importers to import dependencies in the same way that it did before.